### PR TITLE
update sql rules

### PR DESCRIPTION
**Pull request by LewisArdern**
* Improved taint for express using typescript and js sources
* Updated the source to come from a function argument for SQLI since thats pretty much what the rule is doing the other rule would flag any type of issue which would add negative develop sentiment

### CODE REVIEW

**Review by inkz — APPROVED** [review]
good!

File: javascript/express/security/injection/tainted-sql-string.yaml [hunk 1]
```diff
@@ -20,34 +21,56 @@ rules:
...
+              - pattern: $REQ.headers
+      - patterns:
+          - pattern-either:
+              - pattern-inside: >
+                  ({ $REQ }: Request,$RES: Response, $NEXT: NextFunction) =>
+                  {...}
+              - pattern-inside: |
+                  ({ $REQ }: Request,$RES: Response) => {...}
+          - pattern-either:
+              - pattern: params
```
Comment (IagoAbal) [inline javascript/express/security/injection/tainted-sql-string.yaml:53]: Don't you want to make sure that `params` etc are values of `$REQ` ?

File: javascript/express/security/injection/tainted-sql-string.yaml [hunk 2]
```diff
@@ -20,34 +21,56 @@ rules:
...
                   - pattern-either:
-                      - pattern: |
+                      - pattern-inside: |
                           "$SQLSTR" + $EXPR
-                      - pattern: |
-                          "$SQLSTR".concat(...)
-                      - pattern: util.format($SQLSTR, ...)
+                      - pattern-inside: |
+                          "$SQLSTR".concat($EXPR)
+                      - pattern: util.format($SQLSTR, $EXPR)
```
Comment (IagoAbal) [inline javascript/express/security/injection/tainted-sql-string.yaml:67]: It's strange to mix `pattern-inside` and `pattern` in a `pattern-either`. Is that what you want?

File: javascript/express/security/injection/tainted-sql-string.yaml [hunk 3]
```diff
@@ -20,34 +21,56 @@ rules:
...
                   - metavariable-regex:
                       metavariable: $SQLSTR
                       regex: .*\b(?i)(select|delete|insert|create|update|alter|drop)\b.*
               - patterns:
                   - pattern: |
-                      `...${...}...`
+                      `...${$EXPR}...`
                   - pattern-regex: |
                       .*\b(?i)(select|delete|insert|create|update|alter|drop)\b.*
```
Comment (IagoAbal) [inline javascript/express/security/injection/tainted-sql-string.yaml:75]: Is this regex meant to only apply inside
```
`...${$EXPR}...`
```
?

File: javascript/lang/security/audit/sqli/node-mysql-sqli.yaml [hunk 4]
```diff
@@ -1,66 +1,54 @@
...
+      technology:
+        - mysql
+        - mysql2
+        - javascript
+        - nodejs
+      license: Commons Clause License Condition v1.0[LGPL-2.1-only]
+    pattern-sources:
+      - patterns:
+          - pattern-inside: function ... (..., $Y,...) {...}
+          - pattern: $Y
```
Comment (IagoAbal) [inline javascript/lang/security/audit/sqli/node-mysql-sqli.yaml:32]: `focus-metavariable` is better here

File: javascript/express/security/injection/tainted-sql-string.yaml (hunk 3 above)
Comment (LewisArdern) [reply javascript/express/security/injection/tainted-sql-string.yaml:75]: No, its meant to be most likely at the start of the expression, but the 'source should be $EXPR

File: javascript/lang/security/audit/sqli/node-mysql-sqli.yaml (hunk 4 above)
Comment (LewisArdern) [reply javascript/lang/security/audit/sqli/node-mysql-sqli.yaml:32]: noted! i will update

**Pull request merged**